Simplify example for nn.Flatten

mdmittriy opened the pull request:
Fixes #67415
Using the docstring example provided by @jbschlosser to the issue submitted by @qzylalala

Commit:
Issue fix #67415

jbschlosser reviewed (commented):
Thanks for the fix! Do you mind adding another example with `start_dim` and `end_dim` set to non-default values?

Commit:
Added one more example with non-default parameters

Added another example with start_dim and end_dim set to non-default values and
some comments

Commit:
Changed spacing to be coherent with other examples

jbschlosser reviewed (approved):
LGTM! Thanks for the update :)
